examples/gno.land/p/demo/gnolotto/gnolotto.gno

```diff
@@ -0,0 +1,113 @@
+package gnolotto
+
+import (
+	"std"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Ticket struct {
+	Numbers []int       // Holds the selected numbers for the lottery ticket
+	Owner   std.Address // Address of the ticket owner
+}
+
+type Lottery struct {
+	Tickets        []Ticket  // All tickets in the lottery
+	WinningNumbers []int     // Winning numbers after the draw
+	DrawTime       time.Time // Time of the draw
+	PrizePool      int64     // Total prize pool amount
+}
+
+// Intializes a new lottery instance with a specified draw time and prize pool
+func NewLottery(drawTime time.Time, prizePool int64) *Lottery {
+	return &Lottery{
+		DrawTime:  drawTime,
+		PrizePool: prizePool,
+		Tickets:   make([]Ticket, 0),
+	}
+}
+
+const MaxLottoNumbers = 5
+
+// Adds a new ticket to the lottery
+func (l *Lottery) AddTicket(numbers []int, owner std.Address) {
+	l.Tickets = append(l.Tickets, Ticket{Numbers: numbers, Owner: owner})
+}
+
+// Conducts the draw by generating 5 pseudo-random numbers between 1 and 15 inclusive
+func (l *Lottery) Draw() {
+	var blockHeight int64 = std.GetHeight()
+
+	l.WinningNumbers = nil
+	numbersMap := make(map[int]bool)
+
+	// Add variability to the pseudo-random number generation
+	var variabilityFactor int64 = 1
+
+	for len(l.WinningNumbers) < MaxLottoNumbers {
+		simpleSeed := (blockHeight + variabilityFactor*251) % 233280
+		number := int(simpleSeed%15) + 1 // Ensure number is between 1 and 15
+
+		if !numbersMap[number] {
+			l.WinningNumbers = append(l.WinningNumbers, number)
+			numbersMap[number] = true
+		}
+
+		variabilityFactor += 13 // Adjusts for increased variability
+	}
+}
```

Comment:
Now that we have `p/demo/entropy`, let's use that. The main reason someone might check this package out in the examples folder is to check how randomness is done (lottery = random).